clients/src/main/java/org/apache/kafka/common/requests/FetchRequest.java

```diff
@@ -296,11 +276,23 @@ public AbstractResponse getErrorResponse(int throttleTimeMs, Throwable e) {
         // may not be any partitions at all in the response.  For this reason, the top-level error code
         // is essential for them.
         Errors error = Errors.forException(e);
-        LinkedHashMap<TopicPartition, FetchResponseData.PartitionData> responseData = new LinkedHashMap<>();
-        for (Map.Entry<TopicPartition, PartitionData> entry : fetchData.entrySet()) {
-            responseData.put(entry.getKey(), FetchResponse.partitionResponse(entry.getKey().partition(), error));
+        List<FetchResponseData.FetchableTopicResponse> topicResponseList = new ArrayList<>();
+        // For version 13+, we know the client can handle a top level error code, so we don't need to send back partitions too.
```

Comment:
Could we adjust the above comment on "The error is indicated in two ways: by setting the same error code in all partitions, and by setting the top-level error code.  The form where we set the same error code in all partitions is needed in order to maintain backwards compatibility with older versions of the protocol in which there was no top-level error code." ?

Reply:
We should adjust this to say we will no longer set on all partitions for versions 13+?

Reply:
right